test/jdk/java/net/httpclient/HttpResponseConnectionLabelTest.java

```diff
@@ -0,0 +1,441 @@
+                                LOGGER.log(
+                                        "Server[%s] has failed! (connectionKey=%s, responseBody=%s)",
+                                        serverId, connectionKey, responseBody);
+                                ioe.printStackTrace(System.out);
+                            }
+                        } catch (InterruptedException _) {
+                            Thread.currentThread().interrupt(); // Restore the interrupt
+                        }
+                    },
+                    handlerPath);
+
+            // Create the client and the request
+            HttpRequest request = HttpRequest.newBuilder(requestUri).version(version).build();
+
+            // Create the pair
+            return new ServerRequestPair(server, request, secure, serverResponseLatchRef);
+
+        }
+
+        private static HttpTestServer createServer(HttpClient.Version version, SSLContext sslContext) {
+            try {
+                return HttpTestServer.create(version, sslContext, ForkJoinPool.commonPool());
+            } catch (IOException exception) {
+                throw new UncheckedIOException(exception);
+            }
+        }
+
+        @Override
+        public String toString() {
+            String version = server.getVersion().toString();
+            return secure ? version.replaceFirst("_", "S_") : version;
+        }
+
+    }
+
+    @AfterAll
+    static void closeServers() {
+        Exception[] exceptionRef = {null};
+        Stream
+                .of(PRI_HTTP1, PRI_HTTPS1, PRI_HTTP2, PRI_HTTPS2, SEC_HTTP1, SEC_HTTPS1, SEC_HTTP2, SEC_HTTPS2)
+                .forEach(pair -> {
+                    try {
+                        pair.server.stop();
+                    } catch (Exception exception) {
+                        if (exceptionRef[0] == null) {
+                            exceptionRef[0] = exception;
+                        } else {
+                            exceptionRef[0].addSuppressed(exception);
+                        }
+                    }
+                });
+        if (exceptionRef[0] != null) {
+            throw new RuntimeException("failed closing one or more server-request pairs", exceptionRef[0]);
+        }
+    }
+
+    static ServerRequestPair[] testParallelRequestsToSameServer() {
+        return new ServerRequestPair[]{
+                PRI_HTTP1,
+                PRI_HTTPS1,
+                PRI_HTTP2,
+                PRI_HTTPS2
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("testParallelRequestsToSameServer")
+    void testParallelRequestsToSameServer(ServerRequestPair pair) throws Exception {
+
+        // There is no implementation-agnostic reliable way to force admission
+        // of multiple connections targeting the same server to an HTTP/2 pool.
+        if (HttpClient.Version.HTTP_2.equals(pair.server.getVersion())) {
+            return;
+        }
```

Comment:
Instead you could send a first request to warm-up the client/server and get a connection in the pool.
Then you could assert that if the version is HTTP/2, the connectionLabels are identical (instead of different)

Reply:
In that case, it would not be a _parallel_, but a _sequential_ test, and that scenario is already tested in `testSerialRequestsToSameServer`.